packages/xstate-react/README.md

```diff
@@ -221,6 +221,21 @@ const App = ({ service }) => {
 };
 ```
 
+When the selector depends on a prop, it must be memoized with `React.useCallback()`:
+
+```js
+import { useSelector } from '@xstate/react';
+
+const App = ({ service, offset }) => {
+  const count = useSelector(
+    service,
+    React.useCallback((state) => state.context.count + offset, [offset])
+  );
+
+  // ...
+};
+```
```

Comment:
Wouldn't this be better done as this?

```js
const firstcount = useSelector(/* ... */);
const count = useMemo(() => firstCount + offset, [firstCount, offset]);
```

Reply:
Either variation is valid. This however exemplifies that if the selector needs to change for some reason, it needs to be wrapped in a `React.useCallback()`. We can't know for sure which approach will be chosen by users, so it makes sense to call attention to the fact that functions that change their return value must be memorized with `React.useCallback()`.